packages/gateway/src/durable-objects/summary-metrics.js

```diff
@@ -188,57 +201,75 @@ export class SummaryMetrics0 {
         RESPONSE_TIME_HISTOGRAM_ID,
         this.responseTimeHistogram
       ),
+      this.state.storage.put(
+        TOTAL_RESPONSES_BY_QUERY_TYPE_ID,
+        this.totalResponsesByQueryType
+      ),
     ])
   }
 
   /**
    * @param {FetchStats} stats
    */
-  _updateContentLengthMetrics(stats) {
+  _updateContentMetrics(stats) {
+    // Content Length
     this.totalContentLengthBytes += BigInt(stats.contentLength)
-
-    // Update histogram
-    const tmpHistogram = {
-      ...this.contentLengthHistogram,
-    }
-
-    // Get all the histogram buckets where the content size is smaller
-    const histogramCandidates = contentLengthHistogram.filter(
-      (h) => stats.contentLength < h
+    this.contentLengthHistogram = getUpdatedHistogram(
+      this.contentLengthHistogram,
+      contentLengthHistogram,
+      stats.contentLength
     )
-    histogramCandidates.forEach((candidate) => {
-      tmpHistogram[candidate] += 1
-    })
 
-    this.contentLengthHistogram = tmpHistogram
+    // Query type
+    if (stats.pathname && stats.pathname !== '/') {
+      this.totalResponsesByQueryType['CID+PATH'] += 1
+    } else {
+      this.totalResponsesByQueryType['CID'] += 1
+    }
   }
 
   /**
    * @param {FetchStats} stats
    */
   _updateResponseTimeHistogram(stats) {
-    const tmpHistogram = {
-      ...this.responseTimeHistogram,
-    }
-
-    // Get all the histogram buckets where the response time is smaller
-    const histogramCandidates = responseTimeHistogram.filter(
-      (h) => stats.responseTime < h
+    this.responseTimeHistogram = getUpdatedHistogram(
+      this.responseTimeHistogram,
+      responseTimeHistogram,
+      stats.responseTime
     )
+  }
+}
 
-    histogramCandidates.forEach((candidate) => {
-      tmpHistogram[candidate] += 1
+function getUpdatedHistogram(histogramData, histogramBuckets, value) {
```

Comment:
Apologies for adding this refactor here 😬  I did the code because this PR already had MimeTypes histogram, but now I think we should keep it as it simplifies current histograms and if we add more in the future